lib/jsonpatch/utils.ex

```diff
@@ -0,0 +1,349 @@
+        end
+    end
+  end
+
+  @doc """
+  Uses a JSON patch path to get the last map/list that this path references.
+
+  ## Examples
+
+      iex> path = "/a/b/c/d"
+      iex> target = %{"a" => %{"b" => %{"c" => %{"d" => 1}}}}
+      iex> Jsonpatch.Utils.get_destination(target, path)
+      {:ok, {%{"d" => 1}, "d"}}
+
+      iex> # Invalid path
+      iex> path = "/a/e/c/d"
+      iex> target = %{"a" => %{"b" => %{"c" => %{"d" => 1}}}}
+      iex> Jsonpatch.Utils.get_destination(target, path)
+      {:error, {:invalid_path, ["a", "e"]}}
+
+      iex> path = "/a/b/1/d"
+      iex> target = %{"a" => %{"b" => [true, %{"d" => 1}]}}
+      iex> Jsonpatch.Utils.get_destination(target, path)
+      {:ok, {%{"d" => 1}, "d"}}
+
+      iex> path = "/a/b/1"
+      iex> target = %{"a" => %{"b" => [true, %{"d" => 1}]}}
+      iex> Jsonpatch.Utils.get_destination(target, path)
+      {:ok, {[true, %{"d" => 1}], 1}}
+
+      iex> # Invalid path
+      iex> path = "/a/b/42/d"
+      iex> target = %{"a" => %{"b" => [true, %{"d" => 1}]}}
+      iex> Jsonpatch.Utils.get_destination(target, path)
+      {:error, {:invalid_path, ["a", "b", "42"]}}
+  """
+
+  @spec get_destination(
+          target :: Types.json_container(),
+          path :: String.t(),
+          Types.opts()
+        ) ::
+          {:ok, {Types.json_container(), last_fragment :: Types.casted_fragment()}}
+          | Types.error()
+  def get_destination(target, path, opts \\ []) do
+    with {:ok, fragments} <- split_path(path) do
+      find_destination(target, [], fragments, opts)
+    end
+  end
+
+  @doc """
+  Updatest a map/list reference by a given JSON patch path with the new destination.
+
+  ## Examples
+
+      iex> path = "/a/b/c/d"
+      iex> target = %{"a" => %{"b" => %{"c" => %{"xx" => 0, "d" => 1}}}}
+      iex> Jsonpatch.Utils.update_destination(target, %{"e" => 1}, path)
+      {:ok, %{"a" => %{"b" => %{"c" => %{"e" => 1}}}}}
+
+      iex> path = "/a/b/1"
+      iex> target = %{"a" => %{"b" => [0, 1, 2]}}
+      iex> Jsonpatch.Utils.update_destination(target, 9999, path)
+      {:ok, %{"a" => %{"b" => 9999}}}
+  """
+  @spec update_destination(
+          target :: Types.json_container(),
+          value :: term(),
+          String.t(),
+          Types.opts()
+        ) ::
+          {:ok, Types.json_container()} | Types.error()
+  def update_destination(target, value, path, opts \\ []) do
+    with {:ok, fragments} <- split_path(path) do
+      do_update_destination(target, value, [], fragments, opts)
+    end
+  end
+
+  @doc """
+  Unescape `~1` to  `/` and `~0` to `~`.
+  """
+  @spec unescape(fragment :: String.t() | integer()) :: String.t()
+  def unescape(fragment) when is_binary(fragment) do
+    fragment
+    |> String.replace("~0", "~")
+    |> String.replace("~1", "/")
+  end
+
+  @doc """
+  Escape `/` to `~1 and `~` to `~0`.
+  """
+  @spec escape(fragment :: String.t() | integer()) :: String.t()
+  def escape(fragment) when is_binary(fragment) do
```

Comment:
Here also, the `escape` should contain a check if the replacing is necessary. See https://github.com/corka149/jsonpatch/issues/11